### PR TITLE
Moved apps and events from ZeroEQHBP vocabulary, add Request event for existing request-reply usage

### PR DESCRIPTION
**Pull request by dnachbaur**

### CODE REVIEW

File: apps/sendevent.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+void parseScript( const char* filename, Events& events );
+
+int main( int argc, char** argv )
+{
+    parseArguments( argc, argv );
+
+    zeroeq::Publisher publisher;
+    Events events;
+    parseScript( scriptFile, events );
+    for( Events::const_iterator i = events.begin(); i != events.end(); ++i )
```
Comment (hernando) [inline apps/sendevent.cpp]: C++11 foreach?

File: apps/sendevent.cpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+ZeroBufPtr parseCellSetBinaryOp( std::istream& input )
+{
+    uint32_ts first;
+    uint32_ts second;
+    lexis::data::CellSetBinaryOpType operation;
+
+    if( !parseUint32_ts( input, first ))
+    {
+        std::cerr << "Error parsing CELLSETBINARYOP parameter 1" << std::endl;
```
Comment (hernando) [inline apps/sendevent.cpp]: s/CELLSETBINARYOP/lexis::data::CellSetBinaryOp/

File: apps/sendevent.cpp [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    return ZeroBufPtr( new lexis::data::CellSetBinaryOp( first, second, operation ));
+}
+
+ZeroBufPtr parseToggleIDRequest( std::istream& input )
+{
+    uint32_ts ids;
+
+    if( !parseUint32_ts( input, ids ))
+    {
+        std::cerr << "Error parsing TOGGLEIDREQUEST parameter" << std::endl;
```
Comment (hernando) [inline apps/sendevent.cpp]: s/TOGGLEIDREQUEST/lexis::data::ToggleIDRequest/

File: apps/sendevent.cpp [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        float pause = 0;
+        // The pause parameter is optional, if the extraction fails it will
+        // be zero
+        buffer >> eventName >> pause;
+        if( eventName == lexis::data::CellSetBinaryOp::ZEROBUF_TYPE_NAME( ))
+        {
+            events.push_back( std::make_pair( pause,
+                                              parseCellSetBinaryOp( input )));
+        }
+        else if ( eventName == lexis::data::ToggleIDRequest::ZEROBUF_TYPE_NAME( ))
```
Comment (hernando) [inline apps/sendevent.cpp:156]: Don't you want to add the seleted ids event?

**Comment by hernando** [comment]
I like the simplicity of the commit compared to the original code. Despite all the criticisms I think that ZeroBuf is doing a good job here.

**Comment by hernando** [comment]
The tests are missing. Or is it enough with the coverage testing in ZeroBuf itself?

File: apps/sendevent.cpp (hunk 4 above)
Comment (dnachbaur) [inline apps/sendevent.cpp:156]: Added it.

**Comment by dnachbaur** [comment]
Retest this please

**Comment by hernando** [comment]
+1

**Pull request merged**